File: chapters/06_workspace_memory_and_search/v1_hardcoded_knowledge.ts
// TypeScript mirror for Chapter 06: Workspace Memory and Search (v1)
// Run with: npx ts-node v1_hardcoded_knowledge.ts

const KNOWLEDGE: Record<string, string> = {
  session: "OpenClaw supports configurable session scoping.",
  queue: "OpenClaw uses queue lanes for controlled concurrency.",
  sandbox: "OpenClaw enforces sandbox boundaries for safe tool execution.",
};

function answerHardcoded(query: string): string {
  const lowered = query.toLowerCase();
  for (const [key, value] of Object.entries(KNOWLEDGE)) {
    if (lowered.includes(key)) {
      return value;
    }
  }
  return "No hardcoded memory hit.";
}

function main(): void {
  console.log(answerHardcoded("How does session work?"));
}

main();

File: chapters/06_workspace_memory_and_search/v1_hardcoded_knowledge.py
from __future__ import annotations


KNOWLEDGE = {
    "session": "OpenClaw supports configurable session scoping.",
    "queue": "OpenClaw uses queue lanes for controlled concurrency.",
    "sandbox": "OpenClaw enforces sandbox boundaries for safe tool execution.",
}


def answer_hardcoded(query: str) -> str:
    lowered = query.lower()
    for key, value in KNOWLEDGE.items():
        if key in lowered:
            return value
    return "No hardcoded memory hit."


def main() -> None:
    print(answer_hardcoded("How does session work?"))


if __name__ == "__main__":
    main()
